File: Database-Population/imdb_fake/celebrities.py
import mysql.connector
from datetime import datetime


def displayTime():
    now = datetime.now()
    current_time = now.strftime("%H:%M:%S")
    print(current_time)


insert_celeb = """insert into celebrities (celebrity_id, celebrity_name, birth_year, death_year)
                    values ('{}', '{}', {}, {})"""

celeb_info = open(r"D:\workspace\IMDb data\celebrities.tsv", "r", encoding='utf-8')

db = mysql.connector.connect(
    host='localhost', user='root', password='', database='imdb_fake')

cursor = db.cursor()

if __name__ == '__main__':

    cursor.execute("select * from professions")
    prof = cursor.fetchall()

    professions_dict = {}

    for i in range(len(prof)):
        professions_dict[prof[i][1]] = prof[i][0]

    line = None
    celeb_data = None

    for i in range(2):
        line = celeb_info.readline()
        celeb_data = line.strip('\n').split('\t')
        print(celeb_data)

    displayTime()

    print(professions_dict)

    count = 0

    while line:

        count += 1
        
        if count % 10000 == 0:
            displayTime()
            print(count)
        
        celeb_id = celeb_data[0]
        celeb_name = celeb_data[1].replace("'", "\\'")

        birth = celeb_data[2] if celeb_data[2] != '\\N' else None
        death = celeb_data[3] if celeb_data[3] != '\\N' else None

        try:
            if birth and death:
                cursor.execute(insert_celeb.format(celeb_id, celeb_name, birth, death))
            elif birth:
                cursor.execute(
                    "insert into celebrities (celebrity_id, celebrity_name, birth_year) values ('{}', '{}', {})"
                    .format(celeb_id, celeb_name, birth))
            elif death:
                cursor.execute(
                    "insert into celebrities (celebrity_id, celebrity_name, death_year) values ('{}', '{}', {})"
                    .format(celeb_id, celeb_name, death))
            else:
                cursor.execute(
                    "insert into celebrities (celebrity_id, celebrity_name) values ('{}', '{}')"
                    .format(celeb_id, celeb_name))
        except mysql.connector.Error as err:
            print(err)
            line = celeb_info.readline()
            celeb_data = line.strip('\n').split('\t')
            continue
        '''
        if celeb_data[4] != '\\N' and celeb_data[4] != '':
            try:
                professions = [professions_dict[x.replace("_", " ").title()] for x in celeb_data[4].split(',')]
            except:
                print(celeb_data[4])
                exit()
            insert_query = "insert into professions_of_celebrities (celebrity_id, profession_id) values "

            for i in range(len(professions) - 1):
                insert_query += "('" + celeb_id + "', " + str(professions[i]) + "), "

            if len(professions) > 0:
                insert_query += "('" + celeb_id + "', " + str(professions[len(professions) - 1]) + ");"
                
            try:
                cursor.execute(insert_query)
            except mysql.connector.Error as err:
                print(err)

        if celeb_data[5] != '\\N' and celeb_data[5] != '':
            knownFor = [x for x in celeb_data[5].split(',')]

            insert_query = "insert into known_for (celebrity_id, title_id) values "

            for i in range(len(knownFor) - 1):
                insert_query += "('" + celeb_id + "', '" + str(knownFor[i]) + "'), "

            if len(knownFor) > 0:
                insert_query += "('" + celeb_id + "', '" + str(knownFor[len(knownFor) - 1]) + "');"
                
            try:
                cursor.execute(insert_query)
            except mysql.connector.Error as err:
                print(err)
        '''
        line = celeb_info.readline()
        celeb_data = line.strip('\n').split('\t')

    db.commit()
    cursor.close()
    db.close()

    displayTime()
    print(count)

File: Database-Population/populate_imdb_fake_db/src/main/java/populate/Celebrities.java
package populate;

import utilities.DatabaseHandling;

import java.sql.Connection;
import java.sql.Statement;
import java.util.Scanner;

import static populate.Populate.displayTime;

public class Celebrities {

    public static void fetchCelebrities(Connection con, Statement statement) {

        try {

            Scanner scanner = DatabaseHandling.getScanner("D:\\workspace\\IMDb data\\celebrities.tsv");

            String[] line = scanner.nextLine().split("\t");

            for(String word : line) {

                System.out.print(word + ' ');
            }

            System.out.println();

            displayTime();

            int count = 0;
            while(scanner.hasNextLine()) {

                count++;

                if(count % 100000 == 0) {

                    System.out.println(count);
                    displayTime();
                    con.commit();
                }

                line = scanner.nextLine().split("\t");

                String title_id = line[0];
                String title_name = line[1].replace("'", "\\'");

                Integer birth = null;
                Integer death = null;

                if(!line[2].equals("\\N") && !line[2].equals("")) {
                    try {
                        birth = Integer.parseInt(line[2]);
                    } catch (Exception e) {
                    }
                }

                if(!line[3].equals("\\N") && !line[3].equals("")) {
                    try {
                        death = Integer.parseInt(line[3]);
                    } catch (Exception e) {
                    }
                }

                String insert_query;
                if(birth != null && death != null){

                    insert_query =
                            "insert into celebrities (celebrity_id, celebrity_name, birth_year, death_year) values ('%s', '%s', %s, %s)";
                    try {
                        statement.execute(String.format(insert_query, title_id, title_name, birth, death));
                    } catch (Exception e) {
                        System.out.println(title_name);
                        System.out.println(String.format(insert_query, title_id, title_name, birth, death));
                        e.printStackTrace();
                    }
                } else if(birth != null) {
                    insert_query =
                            "insert into celebrities (celebrity_id, celebrity_name, birth_year) values ('%s', '%s', %s)";
                    try {
                        statement.execute(String.format(insert_query, title_id, title_name, birth));
                    } catch (Exception e) {
                        System.out.println(title_name);
                        System.out.println(String.format(insert_query, title_id, title_name, birth, death));
                        e.printStackTrace();
                    }
                } else if(death != null) {
                    insert_query =
                            "insert into celebrities (celebrity_id, celebrity_name, death_year) values ('%s', '%s', %s)";
                    try {
                        statement.execute(String.format(insert_query, title_id, title_name, death));
                    } catch (Exception e) {
                        System.out.println(title_name);
                        System.out.println(String.format(insert_query, title_id, title_name, birth, death));
                        e.printStackTrace();
                    }
                } else {
                    insert_query =
                            "insert into celebrities (celebrity_id, celebrity_name) values ('%s', '%s')";

                    try {
                        statement.execute(String.format(insert_query, title_id, title_name));
                    } catch (Exception e) {
                        System.out.println(title_name);
                        System.out.println(String.format(insert_query, title_id, title_name, birth, death));
                        e.printStackTrace();
                    }
                }
            }

            con.commit();
            statement.close();
            con.close();

        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
